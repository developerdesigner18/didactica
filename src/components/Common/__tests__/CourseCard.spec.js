import { createLocalVue, mount } from "@vue/test-utils";
import { waitRAF } from "@/../tests/utils.js";

import LxpCourseCard from "../CourseCard.vue";
import { BImg } from "bootstrap-vue";

// create an extended `Vue` constructor
const localVue = createLocalVue();

localVue.component("lxp-course-card", LxpCourseCard);
localVue.component("b-img", BImg);

describe("CourseCard.vue", () => {
  it("render when created ", async () => {
    let wrapper = await getWrapper();
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
    wrapper.destroy();
  });
  describe("Props", () => {
    let wrapper;

    afterEach(() => {
      wrapper.destroy();
    });

    it("should show tag content if `tagText` is present", async () => {
      wrapper = await getWrapper();
      await wrapper.setProps({
        tagText: "New"
      });
      await waitRAF();

      const cardTag = wrapper.find(".card-tag");
      expect(cardTag.text()).toEqual("New");
    });
  });

  describe("Events", () => {
    let wrapper;

    beforeEach(async () => {
      wrapper = await getWrapper();
    });

    afterEach(() => {
      wrapper.destroy();
    });

    it("should emit `card-click` when card is clicked", async () => {
      let courseCard = wrapper.find(".lxp-course-card");
      await courseCard.trigger("click");
      expect(wrapper.emitted("card-click")).toBeTruthy();
    });
  });
});

function getWrapper() {
  return mount(LxpCourseCard, {
    propsData: {
      cardTitle: "Card Title",
      cardSubTitle: "Subtitle"
    },
    localVue
  });
}
