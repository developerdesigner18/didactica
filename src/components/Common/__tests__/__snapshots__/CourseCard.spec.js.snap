// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseCard.vue render when created  1`] = `
<div
  class="lxp-course-card"
  title="Card Title"
>
  <div
    class="lxp-course-card--div"
  >
    <div
      class="lxp-course-card__img"
    >
      <img
        alt="course image"
        class="lxp-cc__img"
        src="https://files.lxp.academy.who.int/didactica/assets/images/course-card/course-illustration.svg"
      />
        
      <!---->
    </div>
     
    <div
      class="lxp-course-card__body"
    >
      <div
        class="lxp-course-card__content"
      >
        <div>
          <div
            class="lxp-course-card__sub-title"
          >
            
        Subtitle
      
          </div>
           
          <div
            class="lxp-course-card__title"
          >
            Card Title
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="lxp-course-card__footer"
    />
  </div>
</div>
`;
